### PR TITLE
feat(plasma-hope): Handling "Select" scrollbar

### PR DESCRIPTION
**Pull request by Yakutoc**
Необходимо добавить возможность управлять высотой выпадающего меню.

Добавлен новый **API**. 

Новые `props` для компонента `Select`: 

- **listOverflow**
- **listHeight**

```jsx
<Select                 
   listHeight={6}
   listOverflow="scroll" 
/>
```
----------

**Было:**

image

**Стало:** 
Снимок экрана 2023-03-09 в 13 06 12
     

:baby_chick: Download canary assets:

[colors--canary.397.4433611952.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.397.4433611952.xml)  
[color_metro_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.397.4433611952.swift)  
[color_metro_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.397.4433611952.kt)  
[color_metro_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.397.4433611952.ts)  
[color_sberHealth_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.397.4433611952.swift)  
[color_sberHealth_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.397.4433611952.kt)  
[color_sberHealth_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.397.4433611952.ts)  
[color_sbermarket_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.397.4433611952.swift)  
[color_sbermarket_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.397.4433611952.kt)  
[color_sbermarket_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.397.4433611952.ts)  
[color_sberprime_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.397.4433611952.swift)  
[color_sberprime_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.397.4433611952.kt)  
[color_sberprime_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.397.4433611952.ts)  
[color_selgros_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.397.4433611952.swift)  
[color_selgros_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.397.4433611952.kt)  
[color_selgros_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.397.4433611952.ts)  
[color_smbusiness_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.397.4433611952.swift)  
[color_smbusiness_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.397.4433611952.kt)  
[color_smbusiness_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.397.4433611952.ts)  
[PlasmaTokensColor--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.397.4433611952.swift)  
[shadow_sbermarket_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.397.4433611952.ts)  
[typo_mage_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.397.4433611952.swift)  
[typo_mage_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.397.4433611952.kt)  
[typo_mage_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.397.4433611952.ts)  
[typo_plasma_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.397.4433611952.swift)  
[typo_plasma_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.397.4433611952.kt)  
[typo_plasma_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.397.4433611952.ts)  
[typo_ruler_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.397.4433611952.swift)  
[typo_ruler_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.397.4433611952.kt)  
[typo_ruler_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.397.4433611952.ts)  
[typo_sage_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.397.4433611952.swift)  
[typo_sage_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.397.4433611952.kt)  
[typo_sage_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.397.4433611952.ts)  
[typo_sbermarket_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.397.4433611952.swift)  
[typo_sbermarket_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.397.4433611952.kt)  
[typo_sbermarket_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.397.4433611952.ts)  
[typo_soulmate_ios-swift--canary.397.4433611952.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.397.4433611952.swift)  
[typo_soulmate_kotlin--canary.397.4433611952.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.397.4433611952.kt)  
[typo_soulmate_react-native--canary.397.4433611952.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.397.4433611952.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.161.0-canary.397.4433611952.0
  npm install @salutejs/plasma-hope@0.19.0-canary.397.4433611952.0
  npm install @salutejs/plasma-web@1.186.0-canary.397.4433611952.0
  # or 
  yarn add @salutejs/plasma-b2c@1.161.0-canary.397.4433611952.0
  yarn add @salutejs/plasma-hope@0.19.0-canary.397.4433611952.0
  yarn add @salutejs/plasma-web@1.186.0-canary.397.4433611952.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-397/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-397/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-397/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-397/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-397/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-397/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-397/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-397/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-397/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-397/temple-storybook/

File: website/plasma-web-docs/docs/components/Select.mdx [hunk 1]
```diff
@@ -57,3 +57,62 @@ export function App() {
     );
 }
 ```
+
+### Управление высотой выпадающего меню
+Для управления высотой выпадающего меню существуют **2 css custom properties**:
+
+```
+--plasma-dropdown-height
```
Comment (Yeti-or) [inline website/plasma-web-docs/docs/components/Select.mdx:65]: а почему мы не делаем это через props ? у нас нигде нет вроде такого `API`

File: website/plasma-web-docs/docs/components/Select.mdx [hunk 2]
```diff
@@ -57,3 +57,62 @@ export function App() {
     );
 }
 ```
+
+### Управление высотой выпадающего меню
```
Comment (Yeti-or) [inline website/plasma-web-docs/docs/components/Select.mdx:61]: не вижу этого примера что-то https://plasma.sberdevices.ru/pr/pr-395/web/components/select/
может не собралось??

Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Select.mdx:61]: @Yeti-or А вот [тут](https://plasma.sberdevices.ru/pr/pr-397/web/components/select/) есть

Странно. Сейчас пересоберу.

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-397/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-397/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-397/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-397/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-397/temple-storybook/

File: website/plasma-web-docs/docs/components/Select.mdx (hunk 1 above)
Comment (Yakutoc) [reply website/plasma-web-docs/docs/components/Select.mdx:65]: Здесь решил идти по пути наименьшего сопротивления,

Снимок экрана 2023-03-13 в 11 20 31

что бы избежать **prop drill** как на скриншоте. 

``Select -> Select view -> DropdownUncontrolled -> DropdownMenu -> DropdownList ``.     

А так у нас только в компоненте который использует под собой `dropdown` нужно **переопределить** пару значений.

@Yeti-or  Какие ты видишь проблемы с таким подходом?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-397/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-397/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-397/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-397/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-397/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-397/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-397/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-hope/api/plasma-hope.api.md [hunk 3]
```diff
@@ -898,13 +898,13 @@ export const Select: <T>(props: ({
...
...interOverCapture" | "onPointerOut" | "onPointerOutCapture" | "onGotPointerCapture" | "onGotPointerCaptureCapture" | "onLostPointerCapture" | "onLostPointerCaptureCapture" | "onScroll" | "onScrollCapture" | "onWheel" | "onWheelCapture" | "onAnimationStart" | "onAnimationStartCapture" | "onAnimationEnd" | "onAnimationEndCapture" | "onAnimationIteration" | "onAnimationIterationCapture" | "onTransitionEnd" | "onTransitionEndCapture" | "status" | "defaultChecked" | "defaultValue" | "suppressContentEditableWarning" | "suppressHydrationWarning" | "accessKey" | "contentEditable" | "contextMenu" | "dir" | "draggable" | "hidden" | "nonce" | "placeholder" | "spellCheck" | "translate" | "radioGroup" | "about" | "datatype" | "inlist" | "prefix" | "property" | "resource" | "typeof" | "vocab" | "autoCapitalize" | "autoCorrect" | "autoSave" | "itemProp" | "itemScope" | "itemType" | "itemID" | "itemRef" | "results" | "security" | "unselectable" | "inputMode" | "is" | "autoFocus" | "disabled" | "formAction" | "formEncType" | "formMethod" | "formNoValidate" | "formTarget" | "isOpen" | "items" | "onItemSelect" | "helperText"> & DropdownListCssProperties & React_2.RefAttributes<HTMLButtonElement>) | ({
```
Comment (Yeti-or) [inline packages/plasma-hope/api/plasma-hope.api.md:901]: Меня конечно растраивает такие простыни в нашем api :/ но что с этим делать пока хз

File: packages/plasma-hope/src/components/Select/Select.tsx [hunk 4]
```diff
@@ -29,14 +31,15 @@ export type SelectProps<T = any> = (
       }
 ) & {
     /**
-     * Значение контрола.
+     * Значение control.
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Select/Select.tsx:34]: А чем это лучше ?

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**